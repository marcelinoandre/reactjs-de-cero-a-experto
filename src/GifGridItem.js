import React from 'react';
const GifGridItem = ({ img }) => {
  return (
    <div>
      <img src={img.url} alt={img.alt} />
    </div>
  );
};

export default GifGridItem;
