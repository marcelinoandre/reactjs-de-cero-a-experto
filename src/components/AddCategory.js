import { useState } from 'react';

const AddCategory = ({ setCategores }) => {
  const [category, setCategory] = useState('');

  const handleAdd = e => {
    e.preventDefault();
    if (!category) return console.log('vazio');
    setCategores(categores => [...categores, category]);
    setCategory('');
  };

  return (
    <form onSubmit={handleAdd}>
      <input
        name="category"
        type="text"
        value={category}
        onChange={e => setCategory(e.target.value)}
      />
      <button>Add</button>
    </form>
  );
};

export default AddCategory;
