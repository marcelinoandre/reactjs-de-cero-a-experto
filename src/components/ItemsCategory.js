import { useEffect, useState } from 'react';
import GifGridItem from '../GifGridItem';

const ItemsCategory = ({ categories }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url = `https://api.giphy.com/v1/gifs/search?q=rick+and+morty&limit=10&api_key=7Ktw4O8WmoytakpzYiaJYKaUIX32rQOj`;
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      };
    });
    setImages(gifs);
  };

  return (
    <>
      {images.map(img => (
        <GifGridItem key={img.id} img={img} />
      ))}
    </>
  );
};

export default ItemsCategory;
