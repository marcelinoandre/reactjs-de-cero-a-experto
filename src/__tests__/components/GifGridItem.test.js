import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('GifGridItem', () => {
  const title = 'Title Img';
  const url = 'http://localhost:3000/gif.jpg';
  const wrapper = shallow(<GifGridItem url={url} title={title} />);

  it('should show component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a paragraph with a title', () => {
    const paragrafo = wrapper.find('p');
    expect(paragrafo.text().trim()).toBe(title);
  });

  it('should have  in the image the tag alt equal to the title ', () => {
    const img = wrapper.find('img');
    expect(img.props().alt).toBe(title);
  });

  it('should have the className  "animate__fadeIn" ', () => {
    const div = wrapper.find('div');
    expect(div.props().className.includes('animate__fadeIn')).toBe(true);
  });
});
