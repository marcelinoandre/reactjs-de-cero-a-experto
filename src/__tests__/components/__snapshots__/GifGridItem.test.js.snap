// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should show component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title Img"
    src="http://localhost:3000/gif.jpg"
  />
  <p>
     
    Title Img
     
  </p>
</div>
`;
